Fix tests imports to reference lib, not src + fix code coverage

bajtos opened the pull request:
~~Initially, I thought this may be the reason for wrong code coverage numbers we are observing. Unfortunately it was not, but I think it's still an change that should be made.~~

Fix tests imports to reference lib, not src. With that change in place, we can gather code-coverage data for transpiled .js files in lib/lib6 only. This fixes the problem where code coverage numbers were not including all passes.

Configure an exception for example-codehub which is not tranpiled nor published to npmjs.org.

cc @bajtos @raymondfeng @ritch @superkhau

raymondfeng commented:
Is the proposed change to make sure that tests are running against transpiled js files?

superkhau commented:
>Is the proposed change to make sure that tests are running against transpiled js files even for those from the same module?

I approved assuming this is true.

bajtos commented:
> Is the proposed change to make sure that tests are running against transpiled js files even for those from the same module?

Yes, exactly.

bajtos commented:
So my intuition was correct at the end of the day. After I fixed one forgotten place that was still importing from `src` and fixed `nyc` setup to load coverage data for `.js` files only, the code coverage numbers were up and seem to be correct now.

Here is my understanding how code coverage works now:

 - `nyc` instruments `.js` files in `lib`/`lib6` folder
 - tests are executed via `ts-node`
 - `nyc report` processes coverage data collected for transpiled `.js` files. These files include source-map information, which is recognized by `nyc` and thus the coverage reports lists original `.ts` files.

BTW as a result of this change, `npm test` seems to run much faster than before. I suspect this is caused by `ts-node` having less compilation work to do, since it does not have to compile almost any files from `src` folders now, only `test` files and `example-codehub` sources.

@superkhau @raymondfeng PTAL again.
